src/Microsoft.Data/BaseColumn.cs

```diff
@@ -78,6 +79,14 @@ protected set
 
         public virtual GroupBy GroupBy(int columnIndex, DataFrame parent) => throw new NotImplementedException();
 
+        // Arrow related APIs
+        public virtual Field Field() => throw new NotImplementedException();
```

Comment:
I think things like this should be `protected internal`, like I made them in my IDataView PR. Normal public consumers shouldn't need to see/call these.